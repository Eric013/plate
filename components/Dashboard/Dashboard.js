import { inject, observer } from 'mobx-react'

import { Component } from 'react'
import { Message } from 'semantic-ui-react'

export default inject('GlobalStore')(observer(class extends Component {
  render () {
    return (
      <div>
        <Message
          floating
          success
          visible={this.props.GlobalStore.registerMessage}
          hidden={this.props.GlobalStore.registerMessageHidden}
          onDismiss={this.props.GlobalStore.hideRegisterMessage}
          header={'Hello, ' + this.props.GlobalStore.displayUserName}
          content='This is your dashboard where you can add new plates.'
        />
        <h1>Dashboard</h1>
      </div>
    )
  }
}))
