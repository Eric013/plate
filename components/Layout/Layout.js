import Head from 'next/head'
import Navigation from '../Navigation/Navigation'

export default ({ children, menuOpen, openMenu, closeMenu }) => (
  <div>
    <style jsx global>{`
      * {
        margin: 0;
        box-sizing: border-box;
      }
      body {
        background-color: white;
      }
    `}</style>
    <Head>
      <meta charset='utf-8' />
      <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' />
      <meta name='theme-color' content='#00B5AD' />
      <meta name='apple-mobile-web-app-status-bar-style' content='black-translucent' />
      <link rel='shortcut icon' href='/static/plateLogo.png' />
      <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.7/semantic.css' />
    </Head>
    <Navigation
      menuOpen={menuOpen}
      openMenu={openMenu}
      closeMenu={closeMenu}
    />
    {children}
  </div>
)
