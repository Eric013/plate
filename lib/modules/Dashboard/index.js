import DashboardContainer from "./containers/DashboardContainer";
import DefaultLayout from "../../layouts/DefaultLayout/DefaultLayout";
import Header from "../../components/Header/Header";
import React from "react";

export default () => (
  <DefaultLayout>
    <Header title="Dashboard" />
    <DashboardContainer />
  </DefaultLayout>
);
