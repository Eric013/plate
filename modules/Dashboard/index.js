import React, { Component, PropTypes } from "react";

import { AllPlatesQuery } from "./utils/queries";
import DashboardMenu from "./components/DashboardMenu";
import DefaultLayout from "../DefaultLayout";
import Header from "../../components/Header/Header";
import Loader from "../../components/Loader/Loader";
import NoPlatesFound from "./components/NoPlatesFound";
import Plate from "./components/Plate";
import { graphql } from "react-apollo";

class Dashboard extends Component {
  static propTypes = {
    loading: PropTypes.bool,
    allPlates: PropTypes.array,
    addPlate: PropTypes.func
  };

  render() {
    if (this.props.loading) {
      return <Loader />;
    }

    const plates = this.props.allPlates.map((plate, index) => (
      <div
        key={index}
        className="col-xs-12 col-sm-12 col-md-6 col-lg-3 col-xl-3"
        style={{ marginBottom: 10 }}
      >
        <Plate
          plateId={plate.id}
          name={plate.name}
          description={plate.description}
          cardImage={plate.thumbnail}
          completed={plate.completed}
        />
      </div>
    ));

    return (
      <DefaultLayout>
        <Header title="Dashboard" />
        <div className="container-fluid" style={{ paddingTop: 5 }}>
          <div className="row">
            <div
              className="col-xs-12 col-sm-12 col-md-12 col-lg-12 col-xl-12"
              style={{ marginBottom: 10 }}
            >
              <DashboardMenu addPlate={this.props.addPlate} />
            </div>
          </div>
          <div className="row">
            {this.props.allPlates.length === 0 &&
              <div
                className="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-xl-4 col-lg-offset-4 col-xl-offset-4 col-md-offset-3"
              >
                <NoPlatesFound />
              </div>}
            {plates}
          </div>
        </div>
      </DefaultLayout>
    );
  }
}

export default graphql(AllPlatesQuery, {
  props: ({ data: { loading, allPlates } }) => ({
    loading,
    allPlates
  }),
  options: {
    fetchPolicy: "cache-and-network"
  }
})(Dashboard);
