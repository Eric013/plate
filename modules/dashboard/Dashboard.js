import AddPlateModal from './components/AddPlateModal'
import { Component } from 'react'
import { Grid } from 'semantic-ui-react'
import NoPlatesFound from './components/NoPlatesFound'
import Plate from './components/Plate'
import { observer } from 'mobx-react'
import store from './store/store'

@observer
class Dashboard extends Component {
  handleOpen = () => {
    store.addPlateModalOpen = true
  }

  confirmPlate = (e) => {
    e.preventDefault()

    const plateName = document.querySelector('#plateName').value
    const plateDescription = document.querySelector('#plateDescription').value

    store.addPlateModalOpen = false

    this.addPlate(plateName, plateDescription)
  }

  addPlate = (plateName, plateDescription) => {
    const newPlate = {
      id: store.plateId++,
      plateName,
      plateDescription
    }

    store.plates.push(newPlate)
  }

  removePlate = (id) => {
    store.plates = store.plates.filter(plate => plate.id !== id)
  }

  cancelPlate = () => {
    store.addPlateModalOpen = false
  }

  render() {
    const styles = {
      columnPadding: {
        paddingBottom: 10
      }
    }

    return (
      <Grid padded>
        <Grid.Row>
          <Grid.Column>
            <AddPlateModal
              handleOpen={this.handleOpen}
              modalOpen={store.addPlateModalOpen}
              confirmPlate={this.confirmPlate}
              cancelPlate={this.cancelPlate}
            />
          </Grid.Column>
        </Grid.Row>
        {store.plates.length > 0
          ?
          <Grid.Row>
            {store.plates.map((plate, index) => {
              return (
                <Grid.Column
                  key={index}
                  computer={4}
                  largeScreen={4}
                  tablet={8}
                  mobile={16}
                  style={styles.columnPadding}
                >
                  <Plate
                    key={index}
                    plate={plate}
                    removePlate={this.removePlate}
                  />
                </Grid.Column>
              )
            })}
          </Grid.Row>
          :
          <Grid.Row centered>
            <Grid.Column
              textAlign='center'
              computer={4}
              largeScreen={4}
              tablet={8}
              mobile={16}
              >
              <NoPlatesFound />
            </Grid.Column>
          </Grid.Row>
        }
      </Grid>
    )
  }
}

export default Dashboard
