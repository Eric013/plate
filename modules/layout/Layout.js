import { Menu, Sidebar } from 'semantic-ui-react'
import React, { Component } from 'react'

import AuthenticatedMenu from './components/AuthenticatedMenu'
import Head from 'next/head'
import Navigation from '../navigation/Navigation'
import UnauthenticatedMenu from './components/UnauthenticatedMenu'
import { observer } from 'mobx-react'
import store from './store/store'

const MainLayout = observer(class MainLayout extends Component {
  openSidebar = () => {
    store.sidebarOpen = true
  }

  closeSidebar = () => {
    store.sidebarOpen = false
  }

  render() {
    return (
      <div>
        <style jsx global>{`
          html {
            margin: 0;
            padding: 5px;
            padding-top: 30px;
            font-family: sans-serif;
            background-color: #EFEFEF;
          }
        `}</style>
        <Head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no" />
          <meta name="theme-color" content="#00B5AD" />
          <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
          <link rel="stylesheet" href="https//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.6/semantic.min.css"></link>
          <link rel="shortcut icon" href="/static/plateLogo.png" />
        </Head>
        <Sidebar
          as={Menu}
          animation='overlay'
          width='thin'
          visible={store.sidebarOpen}
          icon='labeled'
          color='teal'
          vertical
          inverted
        >
          {store.loggedIn
            ? <AuthenticatedMenu toggleSidebar={this.closeSidebar} />
            : <UnauthenticatedMenu toggleSidebar={this.closeSidebar} />
          }
        </Sidebar>
        <Sidebar.Pusher>
          <div>
            <Navigation toggleSidebar={this.openSidebar} />
            {this.props.children}
          </div>
        </Sidebar.Pusher>
      </div>
    )
  }
})

export default MainLayout
