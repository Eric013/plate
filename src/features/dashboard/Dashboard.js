import * as firebase from 'firebase'

import React, { Component, PropTypes } from 'react'

import AddPlateModal from './components/addPlateModal/AddPlateModal'
import { Grid } from 'semantic-ui-react'
import NoPlatesFound from './components/noPlatesFound/NoPlatesFound'
import Plate from './components/plate/Plate'
import { browserHistory } from 'react-router'

export default class extends Component {
  static propTypes = {
    plates: PropTypes.array,
    addPlate: PropTypes.func,
    removePlate: PropTypes.func
  }

  constructor(props) {
    super(props)
    this.state = {
      modalOpen: false
    }
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(firebaseUser => {
      if (!firebaseUser) {
        browserHistory.push('/login')
      }
    })
  }

  handleOpen = () => {
    this.setState({
      modalOpen: true
    })
  }

  confirmPlate = (e) => {
    e.preventDefault()

    const plateName = document.querySelector('#plateName').value
    const plateDescription = document.querySelector('#plateDescription').value

    this.setState({
      modalOpen: false
    })

    this.props.addPlate(plateName, plateDescription)
  }

  cancelPlate = () => {
    this.setState({
      modalOpen: false
    })
  }

  render() {
    const styles = {
      columnPadding: {
        paddingBottom: 10
      }
    }

    return (
      <Grid padded>
        <Grid.Row>
          <Grid.Column>
            <AddPlateModal
              handleOpen={this.handleOpen}
              modalOpen={this.state.modalOpen}
              confirmPlate={this.confirmPlate}
              cancelPlate={this.cancelPlate}
            />
          </Grid.Column>
        </Grid.Row>
        {this.props.plates.length > 0
          ?
          <Grid.Row>
            {this.props.plates.map((plate, index) => {
              return (
                <Grid.Column
                  key={index}
                  computer={4}
                  largeScreen={4}
                  tablet={8}
                  mobile={16}
                  style={styles.columnPadding}
                >
                  <Plate
                    key={index}
                    plate={plate}
                    removePlate={this.props.removePlate}
                  />
                </Grid.Column>
              )
            })}
          </Grid.Row>
          :
          <Grid.Row centered>
            <Grid.Column
              textAlign='center'
              computer={4}
              largeScreen={4}
              tablet={8}
              mobile={16}
              >
              <NoPlatesFound />
            </Grid.Column>
          </Grid.Row>
        }
      </Grid>
    )
  }
}
