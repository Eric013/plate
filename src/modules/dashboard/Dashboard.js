import * as firebase from 'firebase'

import React, { Component } from 'react'

import AddPlateModal from './components/addPlateModal/AddPlateModal'
import { Grid } from 'semantic-ui-react'
import NoPlatesFound from './components/noPlatesFound/NoPlatesFound'
import Plate from './components/plate/Plate'
import { browserHistory } from 'react-router'

export default class extends Component {
  constructor() {
    super()
    this.state = {
      modalOpen: false,
      plates: [],
      plateId: 0
    }
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(firebaseUser => {
      if (!firebaseUser) {
        browserHistory.push('/login')
      }
    })
  }

  handleOpen = () => {
    this.setState({
      modalOpen: true
    })
  }

  confirmPlate = (e) => {
    e.preventDefault()

    const plateName = document.querySelector('#plateName').value
    const plateDescription = document.querySelector('#plateDescription').value

    this.setState({
      modalOpen: false
    })

    this.addPlate(plateName, plateDescription)
  }

  addPlate = (plateName, plateDescription) => {
    const newPlate = {
      id: this.state.plateId++,
      plateName,
      plateDescription
    }

    this.state.plates.push(newPlate)
  }

  removePlate = (id) => {
    this.setState({
      plates: this.state.plates.filter(plate => plate.id !== id)
    })
  }

  cancelPlate = () => {
    this.setState({
      modalOpen: false
    })
  }

  render() {
    const styles = {
      columnPadding: {
        paddingBottom: 10
      }
    }

    return (
      <Grid padded>
        <Grid.Row>
          <Grid.Column>
            <AddPlateModal
              handleOpen={this.handleOpen}
              modalOpen={this.state.modalOpen}
              confirmPlate={this.confirmPlate}
              cancelPlate={this.cancelPlate}
            />
          </Grid.Column>
        </Grid.Row>
        {this.state.plates.length > 0
          ?
          <Grid.Row>
            {this.state.plates.map((plate, index) => {
              return (
                <Grid.Column
                  key={index}
                  computer={4}
                  largeScreen={4}
                  tablet={8}
                  mobile={16}
                  style={styles.columnPadding}
                >
                  <Plate
                    key={index}
                    plate={plate}
                    removePlate={this.removePlate}
                  />
                </Grid.Column>
              )
            })}
          </Grid.Row>
          :
          <Grid.Row centered>
            <Grid.Column
              textAlign='center'
              computer={4}
              largeScreen={4}
              tablet={8}
              mobile={16}
              >
              <NoPlatesFound />
            </Grid.Column>
          </Grid.Row>
        }
      </Grid>
    )
  }
}
