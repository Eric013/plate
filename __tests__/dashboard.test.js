/* global it, expect, describe */

import Dashboard from '../pages/dashboard'
import Layout from '../components/Layout/Layout'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Layout>
      <Dashboard />
    </Layout>,
    div)
})

describe('Dashboard snapshot', () => {
  it('renders without crashing!"', () => {
    const component = renderer.create(<Dashboard />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
