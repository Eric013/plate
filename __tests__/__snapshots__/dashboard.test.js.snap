exports[`Dashboard snapshot renders without crashing!" 1`] = `
<div>
  <div
    className="nav-bar"
    data-jsx={2548831492}>
    <div
      className="ui top fixed pointing menu">
      <a
        data-jsx={2548831492}
        href="/"
        onClick={[Function]}>
        <div
          className="link item"
          onClick={[Function]}>
          <img
            data-jsx={2548831492}
            src="/static/plateLogo.png" />
          <span
            className="logo-text"
            data-jsx={2548831492}>
            late
          </span>
        </div>
      </a>
      <div
        className="right menu">
        <a
          className="link item"
          onClick={[Function]}>
          <span
            data-jsx={2548831492}>
            MENU
          </span>
        </a>
      </div>
    </div>
    <div
      data-jsx={1706063930} />
  </div>
  <h1>
    Dashboard
  </h1>
</div>
`;
